lib/Conversion/PTensorToLinalg/PTensorToLinalg.cpp

```diff
@@ -156,6 +153,40 @@ struct ExtractRTensorLowering
   }
 };
 
+/// Convert PTensor's extract_slice to tensor::extract_slice.
+struct ExtractSliceLowering
+    : public ::mlir::OpConversionPattern<::imex::ptensor::ExtractSliceOp> {
+  using OpConversionPattern::OpConversionPattern;
+
+  ::mlir::LogicalResult
+  matchAndRewrite(::imex::ptensor::ExtractSliceOp op,
+                  ::imex::ptensor::ExtractSliceOp::Adaptor adaptor,
+                  ::mlir::ConversionPatternRewriter &rewriter) const override {
+    auto loc = op.getLoc();
+
+    // source and result are expected to be of PTensorType
+    auto srcPtTyp =
+        op.getSource().getType().dyn_cast<::imex::ptensor::PTensorType>();
+    if (!srcPtTyp)
+      return ::mlir::failure();
+    auto srcRtTyp = srcPtTyp.getRtensor();
+    auto retPtTyp = op.getType().dyn_cast<::imex::ptensor::PTensorType>();
+    if (!retPtTyp)
+      return ::mlir::failure();
+    auto retRtTyp = retPtTyp.getRtensor();
+    // extract RankedTensor
+    auto source = rewriter.create<::imex::ptensor::ExtractRTensorOp>(
+        loc, srcRtTyp, op.getSource());
+    // and simply replace with tensor::extract_slice
+    rewriter.replaceOpWithNewOp<::mlir::tensor::ExtractSliceOp>(
```

Comment:
Just FYI: we cannot blindly convert all ptensor/ntensor slices to `tensor.extract_slice` as they are reference types and `tensor` is value type and is immutable and some tensor/linalg transformations rely on that. We need some sort of alias analysis to make sure we have no writes (e.g. setitem) to `tensor.extract_slice` result, but this doesn't implemented in ntensor either yet.

But this PR is still ok to as as starting point.

Reply:
I guess you mean writing to the tensor with operations which are not in the tensor dialect. Is that what you mean?

Reply:
Consider following example:
```
array = numpy.zeros(...)
view = array[1:2]
view[0] = 1
print(array[1])
```

Naive conversion to linalg will produce smth like this
```
%0 = linalg.fill ...
%1 = tensor.extract_slice %0, ...
setitem(%1, ...)
%3 = getitem %0
```
And as tenors assumed immutable, it can be converted to
```
%0 = linalg.fill ...
%1 = linalg.fill ...
setitem(%1, ...)
%3 = getitem %0
```

Reply:
Ok. 
I remember that the docs for tensor.insert_slice gave the impression that it provides inplace updates. It now clearly states that it returns a copy, so that is not a way out either.
How would we guard writes from such undesired optimizations? Is there something like an annotation which we can add?

Reply:
For ntensor I'm now also always convert them, but in the future I'm planning to generate `tensor.extract_slice` only if we can prove it is safe (no writes to result), otherwise it will stay as `ntensor.subview` and will be converted to `memref.subview` later. As I said, it require alias analysis which I need to think more of.